### ISSUE TITLE
The Rust code generated for split does not compile

### ISSUE THREAD

**Issue by mbudiu-vmw** [ryzhyk/differential-datalog#81]
The unmerged tutorial has an example which does not seem to generate correct Rust code.
This is the dl code:
```
extern function split(s: string, sep: string): set<string>
function split_ip_list(x: string): set<string> =
   split(x, " ")
```

The generated code does not compile:
```
fn split_ip_list(x: &String) -> set<String> {
    split(x, (&r###" "###.to_string()))
}
```
since there is no set type in Rust.

**Comment by ryzhyk** [comment]
The set type is special. It must be replace by any type that implements the IntoIterator trait, e.g., Vec

**Comment by mbudiu-vmw** [comment]
The problem is generating Rust code for the `set<string>` type itself.
When you output the type you generate `set<String>`, which seems incorrect.

**Comment by ryzhyk** [comment]
Ok, I get it now. Yes, this is broken. Not sure if there's a good way to fix it...

**Comment by mbudiu-vmw** [comment]
Just generate the IntoIterator as output.

**Comment by ryzhyk** [comment]
Fixed by c9f41f323a470370de62fc5c067b44b21ba167f1
